PR: Update installation docs to reflect need to install correct version of spyder-kernels for 3.3+

CAM-Gerlach opened the pull request:
<!--- Before submitting your pull request, --->
<!--- please complete as much as possible of the following checklist: --->

### Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 3.x)
* [x] Checked your writing carefully for correct English spelling, grammar, etc
* [x] Described your changes and the motivation for them below
* [x] Noted what issue(s) this pull request resolves, creating one if needed



## Description of Changes

<!--- Describe what you've changed and why. --->

Should fix the main issue behind spyder-ide/spyder#7337 Hopefully clarifies the install procedure when cloning from source under Spyder 3.3+. Let me know if I missed anything or something was inaccurate (particularly the install command using ``conda``--will it require ``-c spyder-ide``? I couldn't actually test it since 3.3 doesn't appear to be on ``conda`` yet.


### Issue(s) Resolved

<!--- Pull requests should typically resolve at least one—preferably only one—
<!--- outstanding issue; create a new one if no relevant issue exists.
<!--- List the issue(s) below, in the form "Fixes #1234" . One per line.--->

Fixes #55


<!--- Thanks for your help making Spyder --->
<!--- and its documentation better for everyone! --->

CAM-Gerlach commented:
@ccordoba12

ccordoba12 commented on doc/installation.rst:
```diff
@@ -264,6 +265,19 @@ In summary:
 
+   .. note::
+
+      Following the separation of ``spyder-kernels`` into its own modular package in Spyder 3.3.0, you'll need to have the corresponding version of that package available—``0.x`` for Spyder 3 (``3.x`` branch), and ``1.x`` for Spyder 4 (``master`` branch).
```
I think this is better:

> Following the separation of the code that powers Spyder's consoles into its own modular package (called ``spyder-kernels``) ...

CAM-Gerlach replied:
Thanks; I back-ported the same revised line you requested on the Contributing Guide in spyder-ide/spyder#7418 .

ccordoba12 commented on doc/installation.rst:
```diff
@@ -264,6 +265,19 @@ In summary:
+
+      .. code-block:: bash
+
+         conda install spyder-kernels=1.*
```
This is not possible with `conda` because I instructed Anaconda guys to *not* package spyder-kernels >= 1.0.

So, this needs to be

    pip install --no-deps -U spyder-kernels

CAM-Gerlach replied:
Since the ``1.x`` version is available on ``conda-forge``, would ``conda install -c conda-forge spyder-kernels=1`` be better to suggest? That's how I have it now, and it seems to work just fine so far running from source. Is the risk of problems installing a package from a different channel with conda greater than using ``pip``? 

Seems a bit of inconvenience and potential problems for developers, but I assume its to avoid a (bigger) issue with users accidentally installing the wrong version. If using ``pip``, does the default conda-installed version also then need to be removed, in order to not have two versions?

CAM-Gerlach replied:
Also, once this gets cleared up, the Contributing Guide version needs to be fixed too since the problem wasn't caught there.

CAM-Gerlach replied:
UPDATE: Apparently, the 1.0.1 version is on conda defaults, at least according to my ``conda info spyder-kernels`` command.

ccordoba12 commented on doc/installation.rst:
```diff
@@ -264,6 +265,19 @@ In summary:
+
+         conda install spyder-kernels=1.*
+
+      (and using ``conda install spyder-kernels=0.*`` to switch back to the Spyder 3 version), or by ``clone``-ing the `spyder-kernels git repository`_ to somewhere on your path and checking out the appropriate branch (``0.x`` or ``master``) corresponding to the version of Spyder (3 or 4) you would like to run.
```
Two things here:

1. Since you need to use pip to install spyder-kernels >= 1.0, you also need to use it to install 0.x, so you need to change `conda install spyder-kernels=0.*` for `pip install spyder-kernels==0.*`.
2. You don't achieve anything by simply cloning the spyder-kernels repo and checking out the right branch. You also need to run `pip install -e .` to install it in your env (I told you that before).

CAM-Gerlach replied:
> Since you need to use pip to install spyder-kernels >= 1.0, you also need to use it to install 0.x,

I re-wrote that section to clarify the need to use the same command to switch. Also, in the initial instructions, the whole dependency stack, including spyder-kernels, gets installed via ``conda install spyder``. If ``conda install -c conda-forge spyder-kernels=1`` works in the above block as well as ``pip`` (which it seems to in my limited testing), then it avoids this potential problem.

> You also need to run ``pip install -e``

Yup, back-ported.

> I told you that before

Yes, on my PR *after* this one, spyder-ide/spyder#7418 , that had a downstream version of that text. I hadn't yet back-ported the fixes upstream to here.

CAM-Gerlach commented:
@ccordoba12 Since both versions of Spyder-kernels (1.0.1 and 0.2.4) *are* in the ``main`` defaults-channel repo and the installation command ``conda install spyder-kernels=1.*`` works as originally written, can we just revert back to that, or will ``1.0.1`` be explicitly removed per your instructions? For now, I've changed the command accordingly, so if that won't change in the immediate future then this should be ready to merge on my end since I believe I addressed everything else IIRC; if not, then I can revert or we can handle it differently.

ccordoba12 commented:
> Since both versions of Spyder-kernels (1.0.1 and 0.2.4) are in the main defaults-channel repo

That's only the case for Windows, so please revert your last changes.

CAM-Gerlach commented:
> That's only the case for Windows, so please revert your last changes.

Okay, thanks; dropped the commit changing it. I'll update the contributing guide version too as originally mentioned.

ccordoba12 commented on doc/installation.rst:
```diff
@@ -264,6 +265,25 @@ In summary:
+
+      .. code-block:: bash
+
+         conda install -c conda-forge spyder-kernels=1
```
This is not good because it'd lead to mix packages from conda-forge and defaults. Please change it to

    conda install -c spyder-ide spyder-kernels=1

I'll move the spyder-kernels package to our own channel a soon as I can.

CAM-Gerlach replied:
Okay, sure—that does sound better. I tried that first but it didn't work at least back when I tested it since presumably the packages weren't posted there yet as you mention. Fixed.

ccordoba12 commented on doc/installation.rst:
```diff
@@ -264,6 +265,25 @@ In summary:
+
+      .. code-block:: bash
+
+         pip install --no-deps -U spyder-kernels==1.*
```
Since we're going to use conda to install `spyder-kernels`, this could be simpler:

    pip install spyder-kernels==1.*

CAM-Gerlach replied:
Okay, gotcha. I figured that might be but wasn't 100% sure. Done.

CAM-Gerlach commented:
@ccordoba12 Thanks, changed as requested.

ccordoba12 commented:
@CAM-Gerlach, let us move the spyder-kernels packages to our channel before merging this one.

@dalthviz, what's the command to move packages from conda-forge to our channel?

dalthviz commented:
I think it should be something like 
```
anaconda copy conda-forge/spyder/3.3.0 --to-owner spyder-ide
```
[Here some docs](https://docs.anaconda.com/anaconda-repository/user-guide/tasks/pkgs/copy-pkg)

ccordoba12 commented:
Thanks a lot @dalthviz! The packages are in our channel now.
